checkstyle-tester/diff.groovy

```diff
@@ -628,36 +617,40 @@ def getProjectsStatistic(diffDir) {
     return projectsStatistic
 }
 
-def runMavenExecution(srcDir, excludes, checkstyleConfig,
-                      checkstyleVersion, extraMvnRegressionOptions) {
-    println "Running 'mvn clean' on $srcDir ..."
-    def mvnClean = "mvn -e --no-transfer-progress --batch-mode clean"
-    executeCmd(mvnClean)
-    println "Running Checkstyle on $srcDir ... with excludes {$excludes}"
-    def mvnSite = "mvn -e --no-transfer-progress --batch-mode site " +
-        "-Dcheckstyle.config.location=$checkstyleConfig -Dcheckstyle.excludes=$excludes"
-    if (checkstyleVersion) {
-        mvnSite = mvnSite + " -Dcheckstyle.version=$checkstyleVersion"
+def runCliExecution(allJar, srcDir, excludes, checkstyleConfig, saveDir, continueOnError,
+                      extraRegressionOptions) {
+    def saveLoc = new File(saveDir)
+    if (!saveLoc.exists()) {
+        saveLoc.mkdirs()
     }
-    if (extraMvnRegressionOptions) {
-        mvnSite = mvnSite + " "
 
-        if (!extraMvnRegressionOptions.startsWith("-")) {
-            mvnSite = mvnSite + "-"
+    println "Running Checkstyle CLI on $srcDir ... with excludes {$excludes}"
+    def cliCommand = "java -jar $allJar -c $checkstyleConfig $srcDir -f xml -o " +
+        getOsSpecificPath("$saveDir", "checkstyle-result.xml") +
+        " -e " + getOsSpecificPath("$srcDir", ".git") +
+        " -e " + getOsSpecificPath("$srcDir", ".hg")
```

Comment:
can we skip all hidden folders ?

Reply:
How? We cannot identify what is file/folder unless we scan first. We can't be too global as we sometimes work in hidden folders (`.ci-temp`). Also I think there is some issue with relative paths (checkstyle-local has `../../../` which I am still trying to look into where it may not work correctly.

Reply:
Why don't we just update regression config to have a filter for these directories? I don't like the idea of hardcoding something that is configurable.

Reply:
Do you mean `projects-to-test-on` files? If we do this, we then have to add it for every project.

Reply:
No, in the checkstyle config, by means of BeforeExecutionExclusionFileFilter

Reply:
That is an option, but there is no default configuration file for regression technically. ( see https://github.com/checkstyle/contribution/issues/665 )
It would be up to every config creation to add this.

Reply:
>It would be up to every config creation to add this.

Yes, it should be up to user to decide what files they want to check (or not check). If somebody runs regression testing over these files, and has a bunch of errors, problem is clear. We point to template in docs, this is enough: https://github.com/checkstyle/contribution/tree/master/checkstyle-tester#report-generation

Reply:
filter that we have is working on abslute path, so `\..*` will match to `.ci-temp`.

we can cover this by fansy regexp: `((?!\.ci-temp)\.)+`
https://regex101.com/r/HztiL9/1
https://stackoverflow.com/questions/406230/regular-expression-to-match-a-line-that-doesnt-contain-a-word

Reply:
it think it will work even there is no such folder as ".ci-temp".
so it is config just for our CI execution, in other cases it will not effective.

Reply:
I swapped and made it `-x ((?!\\.ci-temp)\\.)+`.

Reply:
@romani CI failed for this. `checkstyle-tester-diff-groovy-patch` takes place in `.ci-temp`.
https://github.com/checkstyle/contribution/blob/6775a61c51ef392caddd6fef681f5c857bc0220e/.ci/validation.sh#L43-L44

I think we need to keep to explicit excludes.

Reply:
@romani `-x ((?!\.ci\-temp)\.)+` matches everything. I think there is something wrong with your regular expression.

Reply:
ok, not a big deal, we can improve later on.

Reply:
Reverted code.